caikit_computer_vision/data_model/tasks.py

```diff
@@ -0,0 +1,45 @@
+# Copyright The Caikit Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+This module holds the Task definitions for all common NLP tasks
+"""
+
+# Standard
+from typing import Iterable
+
+# Local
+from caikit.core import TaskBase, task
+from .object_detection import ObjectDetectionResult
+from .image_classification import ImageClassificationResult
+
+# TODO - add support for image DM primitives
+@task(
+    required_parameters={"inputs": bytes},
```

Comment:
Hugging Face usually takes a URL or encoded bytes (so it's a string).  Then a little code is needed to decide if it looks like a URL to fetch vs encoded image bytes.   I'm not sure that's the best approach, but for me alignment with HF/transformers is generally a good thing.

Reply:
Thanks for the review @markstur! I agree alignment is good. The image based pipelines can consume both strings and PIL Images (via this preprocessing [function](https://github.com/huggingface/transformers/blob/main/src/transformers/image_utils.py#L256), which standardizes the inputs to PIL images). The reason `bytes` was chosen here is that PIL does provide funcs for to/from bytes, so this felt closer as a primitive, but this is just a placeholder for now.

The actual image data model that I am working on porting over holds a PIL image underneath - when that's ported and modules are added, ideally we'll support the union of things that the pipeline can take and the data model wrapper, or something similar 🤔